### DISCUSSION TITLE
Setting S3 Upload to Public?

### DISCUSSION THREAD

**Discussion by ahinkle** [stefanzweifel/sidecar-browsershot#81] [Q&A]
Hi there,

Thanks for this amazing package - it's very helpful to use in Laravel Vapor!

I'm using a conversion of a page to a PDF. It works great - however, the file is always marked as `private` in AWS. I have the `visibility` on my S3 driver to `public`. Does this package follow the same properties as the S3 config, or would there be room for a third argument to `saveToS3()` to determine the visibility status? Any ideas here?

```php
BrowsershotLambda::url(route('manual-download.show', $this->manual))
    ->showBackground()
    ->margins(20, 0, 20, 0)
    ->waitUntilNetworkIdle()
    ->format($this->format)
    ->saveToS3('downloads/manuals/'.$this->manual->slug.'.pdf');
```

```php
's3' => [
    'driver' => 's3',
    'key' => env('AWS_ACCESS_KEY_ID'),
    'secret' => env('AWS_SECRET_ACCESS_KEY'),
    'region' => env('AWS_DEFAULT_REGION'),
    'bucket' => env('AWS_BUCKET'),
    'url' => env('AWS_URL'),
    'endpoint' => env('AWS_ENDPOINT'),
    'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
    'throw' => false,
    'visibility' => 'public',
],
```

**Comment by stefanzweifel** [comment]
Thanks for the kind words. Always puts a smile on my face to hear others enjoy the package.

I honestly haven't touched the code for this feature of the package in a while, so memory is a bit fuzzy here.

These two places should be responsible for storing files on S3.

- https://github.com/stefanzweifel/sidecar-browsershot/blob/6664ee843b926525577847455c01418f04612e1e/src/BrowsershotLambda.php#L66-L70
- https://github.com/stefanzweifel/sidecar-browsershot/blob/6664ee843b926525577847455c01418f04612e1e/resources/lambda/browsershot.js#L106-L113

As we're using the AWS SDK in node, I'm sure there's a way to pass the visibility to the `s3.putObject()` method. My knowledge of S3 is honestly quite shallow, but I think setting `ACL: 'public-read'` would do the trick here.
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/s3/command/PutObjectCommand/

Do you have the capacity to contribute this?
I think all we would need to do is adding a `acl` key to the `setOption()`-call and change the value depending on the `visibility`-config value of the disk.
And then pass that value to AWS using `event.options.s3.acl`.
